6_nodejs/week2/app.js

```diff
@@ -0,0 +1,87 @@
+import express from "express";
+// import {bodyParser} from 'body-parser'
+import fs from "fs";
+
+const port = process.env.PORT || 3000;
+const data = JSON.parse(fs.readFileSync("./document.json"));
+const app = express();
+
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+// GET /search
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+  if (query) {
+    const filteredData = data.filter((item) => {
+      return Object.values(item).some(
+        (value) => typeof value === "string" && value.includes(query)
+      );
+    });
+    if (filteredData.length === 0) {
+      return res.send("not found");
+    }
+    return res.json({ data: filteredData, message: "ok" });
+  }
+  res.json({ data: data, message: "ok" });
+});
```

Comment:
Good job, as it has only one condition, I would write is as ternary syntax and make it short like following:
```
const { q: query } = req.query;
  const searchResult = query ? data.filter(each => Object.values(each).join(' ').search(query) > -1 ) : data
  
  res.json({ data: searchResult, message: "ok" });
``` 
Notes:
- it's best practice to do the response once at the end
- you don't need to send `no result` message in here, as you send an empty array and the frontend will handle the rest.
- using `search` method in js will be good practice, it returns either the index of search key or -1 as not found